polyfill/lib/calendar.mjs

```diff
@@ -90,10 +90,134 @@ export class Calendar {
+    ({ year, month } = ES.RegulateYearMonthRange(year, month, disambiguation));
+    return new constructor(year, month, this, /* refIsoDay = */ 1);
+  }
+  monthDayFromFields(fields, options, constructor) {
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    const disambiguation = ES.ToTemporalDisambiguation(options);
+    // Intentionally alphabetical
+    let { month, day } = ES.ToRecord(fields, [['day'], ['month']]);
+    ({ month, day } = ES.RegulateMonthDay(month, day, disambiguation));
+    return new constructor(month, day, this, /* refIsoYear = */ 1972);
+  }
+  plus(date, duration, options, constructor) {
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    const disambiguation = ES.ToTemporalDisambiguation(options);
+
+    const { years, months, weeks, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = duration;
+    const { days } = ES.BalanceDuration(
+      duration.days,
+      hours,
+      minutes,
+      seconds,
+      milliseconds,
+      microseconds,
+      nanoseconds,
+      'days'
+    );
+
+    let year = GetSlot(date, ISO_YEAR);
+    let month = GetSlot(date, ISO_MONTH);
+    let day = GetSlot(date, ISO_DAY);
+    ({ year, month, day } = ES.AddDate(year, month, day, years, months, weeks, days, disambiguation));
+    ({ year, month, day } = ES.RegulateDateRange(year, month, day, disambiguation));
+
+    return new constructor(year, month, day, this);
+  }
+  minus(date, duration, options, constructor) {
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    const disambiguation = ES.ToTemporalDisambiguation(options);
+
+    const { years, months, weeks, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = duration;
+    const { days } = ES.BalanceDuration(
+      duration.days,
+      hours,
+      minutes,
+      seconds,
+      milliseconds,
+      microseconds,
+      nanoseconds,
+      'days'
+    );
+
+    let year = GetSlot(date, ISO_YEAR);
+    let month = GetSlot(date, ISO_MONTH);
+    let day = GetSlot(date, ISO_DAY);
+    ({ year, month, day } = ES.SubtractDate(year, month, day, years, months, weeks, days, disambiguation));
+    ({ year, month, day } = ES.RegulateDateRange(year, month, day, disambiguation));
+
+    return new constructor(year, month, day, this);
+  }
+  difference(smaller, larger, options) {
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    const largestUnit = ES.ToLargestTemporalUnit(options, 'days', ['hours', 'minutes', 'seconds']);
+    const { years, months, weeks, days } = ES.DifferenceDate(smaller, larger, largestUnit);
+    const Duration = GetIntrinsic('%Temporal.Duration%');
+    return new Duration(years, months, weeks, days, 0, 0, 0, 0, 0, 0);
+  }
+  year(date) {
+    if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
+    return GetSlot(date, ISO_YEAR);
```

Comment:
Maybe we should read `date.getISOCalendarFields()` here as well, then, to reduce differences between calendars?

Reply:
I might have misunderstood your original remark, but I'm not sure I understand what purpose that would serve. Compared with the old, pre-calendar situation, reading the internal slot directly doesn't introduce any observable accesses other than that of invoking the calendar getter itself. Calling the method means the spec will mandate that an observable call to `getISOCalendarFields()` is interposed every time you evaluate `date.year`.

The call to `getISOCalendarFields()` is unavoidable in userland calendars because they can't read slots, but all of the built-in calendars can at least just read the slots.

Reply:
Right, and I was saying that maybe the built in calendars should also have an observable call to make it consistent with custom or polyfilled calendars. But I don't feel strongly; I was just making an observation.